pkg/gui/controllers/toggle_whitespace_action.go

```diff
@@ -9,6 +10,10 @@ type ToggleWhitespaceAction struct {
 }
 
 func (self *ToggleWhitespaceAction) Call() error {
+	if self.c.CurrentContext().GetKey() == context.PATCH_BUILDING_MAIN_CONTEXT_KEY {
+		return nil
+	}
```

Comment:
I say we stick with a global approach: I can see this ending up in a menu that can be accessed globally, and in that case we'd have a check on context to exclude it from the menu

Reply:
Actually after testing this a bit I think we should raise an error here if we're in the patch building context and we're trying to _enable_ ignoring whitespace because the user ought to know that we explicitly don't allow that.

Reply:
Added the error message in 269483e23c. And I forgot that the same problem applies to the staging panels, so I added those in the same commit as well.